numba/cuda/tests/cudapy/test_compiler.py

```diff
@@ -116,6 +116,13 @@ def f():
         ptx, resty = compile_ptx(f, [], lineinfo=True)
         self.check_line_info(ptx)
 
+    def test_non_void_return_type(self):
+        def f(x, y):
+            return x[0] + y[0]
+
+        with self.assertRaises(TypeError):
```

Comment:
It would be handy also to check the error message, just in case something bizarre happens and causes some other `TypeError`:

```suggestion
        with self.assertRaisesRegex(TypeError, 'must have void return type'):
```